builtin/string.go

```diff
@@ -346,11 +348,19 @@ var _ = method(string_NumberQ, "()")
 
 func string_NumberQ(this Value) Value {
 	// see also Lexer.number
-	return SuBool(numberPat.Matches(ToStr(this)))
+	lexer := lexer.NewLexer(ToStr(this))
+	item := lexer.Next()
+	if item.Token == tok.Add || item.Token == tok.Sub {
+		item = lexer.Next()
+	}
+	if item.Token == tok.Number || item.Token == tok.Eof {
+		return True
+	}
```

Comment:
wouldn't this accept Add, Eof ?

Reply:
yes, which is not what we want. fixing it now

Reply:
I think I fixed it

1. Consume +/- if they exist
2. Consume number (takes care of _, e, .)
3. Match `Eof` after the number to make sure we have gotten the number
4. If not return False